Meta/CMake/libweb_generators.cmake

```diff
@@ -111,14 +111,30 @@ function (generate_js_bindings target)
             list(GET BINDINGS_TYPES ${iter} bindings_type)
             get_property(include_paths DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
             list(TRANSFORM include_paths PREPEND -i)
+
+            # Ninja expects the target name in depfiles to be relative to CMAKE_BINARY_DIR, but ${bindings_src} is
+            # relative to CMAKE_CURRENT_BINARY_DIR. CMake >= 3.20 can do the rewriting transparently (CMP0116).
+            if(CMAKE_GENERATOR MATCHES "^Ninja" AND NOT POLICY CMP0116)
+                # FIXME: Drop this branch for cmake_minimum_required(3.20)
+                get_filename_component(full_path ${bindings_src} ABSOLUTE BASE_DIR ${CMAKE_CURRENT_BINARY_DIR})
+                file(RELATIVE_PATH depfile_target ${CMAKE_BINARY_DIR} ${full_path})
+            else()
+                if (POLICY CMP0116)
+                    cmake_policy(SET CMP0116 NEW)
+                endif()
+                set(depfile_target ${bindings_src})
+            endif()
```

Comment:
Alternatively, maybe the generator could write the path of `depfile_target` relative to the cwd to the depfile, then maybe you wouldn't need this (?)

But as is seems reasonable too.

Reply:
(https://github.com/llvm-mirror/llvm/blob/master/cmake/modules/TableGen.cmake#L15 in case you haven't seen it – seems to do something similar to what you're doing.)